src/components/App/App.tsx

```diff
@@ -1,40 +1,83 @@
 import * as React from "react";
 import * as io from "socket.io-client";
+import * as uuid from "uuid";
 
 import "./app.css";
 
 import { Header } from "../Header/Header";
 import { Main } from "../Main/Main";
 
 interface AppState {
-  socket: SocketIOClient.Socket;
   boardId: string;
+  socket: SocketIOClient.Socket;
+  maxVotes?: number;
+  showVoteLimitAlert: boolean;
+  showResults: boolean;
 }
 
 export class App extends React.Component<{}, AppState> {
   constructor(props: {}) {
     super(props);
 
-    const socket = io.connect(`${window.location.origin}`);
-    const boardId = window.location.pathname.split("/").pop() || "";
+    let serverURL = window.location.origin;
+    let boardId = window.location.pathname.split("/").pop() || "";
+
+    if (!!window.location.port && (window.location.port !== "8000")) {
```

Comment:
This should only be the case if the FE is being run in dev mode which will run the client-side on port 4000.